src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import withFullPolicy, {fullPolicyDefaultProps, fullPolicyPropTypes} from './withFullPolicy';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import compose from '../../libs/compose';
+import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
+import Navigation from '../../libs/Navigation/Navigation';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import styles from '../../styles/styles';
+import TextInputWithLabel from '../../components/TextInputWithLabel';
+import ExpensiPicker from '../../components/ExpensiPicker';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Button from '../../components/Button';
+import FixedFooter from '../../components/FixedFooter';
+import {createPolicyRoom} from '../../libs/actions/Report';
+
+const propTypes = {
+    /** All reports shared with the user */
+    reports: PropTypes.shape({
+        reportName: PropTypes.string,
+        type: PropTypes.string,
+        policyID: PropTypes.string,
+    }).isRequired,
+
+    ...fullPolicyPropTypes,
+
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    betas: [],
+    ...fullPolicyDefaultProps,
+};
+
+class WorkspaceNewRoomPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            roomName: '',
+            policyID: '',
+            visibility: CONST.REPORT.VISIBILITY.RESTRICTED,
+            isLoading: false,
+            error: '',
+            workspaceOptions: [],
+        };
+        this.onWorkspaceSelect = this.onWorkspaceSelect.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+        this.checkAndModifyRoomName = this.checkAndModifyRoomName.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.policies.length === prevProps.policies.length) {
+            return;
+        }
+        const workspaces = _.filter(this.props.policies, policy => policy && policy.type === CONST.POLICY.TYPE.FREE);
+
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({workspaceOptions: _.map(workspaces, policy => ({label: policy.name, key: policy.id, value: policy.id}))});
```

Comment:
I don't think you need to do this here, right? Could this not just be done in the render method? That would avoid your `// eslint-disable-next-line react/no-did-update-set-state` as well.

Reply:
Right, but I wanted to avoid iterating through all of the user's policies during every render. Since there's some regex/processing happening with the roomName every time the user types, I want to keep other side effects to a minimum. 

Having it in `render` would basically mean iterating through all of the user's policies with every keystroke on top of the regex/processing that's happening on roomName. Having it in `componentDidUpdate` means we only iterate when the policies change, which should be relatively infrequently.